File: src/AutomatonTheory/JGraph/DialogBoxes/SetInitialStateDialog.java
package AutomatonTheory.JGraph.DialogBoxes;

import javax.swing.*;

public class SetInitialStateDialog {

    public String[] States;

    public JComboBox<String> initialStateComboBox = new JComboBox<String>();
    public String newInitialState = "";

    public SetInitialStateDialog(String[] states){
        States = states;
    }
    public void displayGUI() {
        JOptionPane.showConfirmDialog(null,
                getPanel(),
                "Set Initial State",
                JOptionPane.OK_CANCEL_OPTION,
                JOptionPane.PLAIN_MESSAGE);
        newInitialState = initialStateComboBox.getSelectedItem().toString();
    }

    private JPanel getPanel(){

        JPanel panel = new JPanel();
        panel.setLayout(new BoxLayout(panel, BoxLayout.Y_AXIS));
        JLabel originLabel = new JLabel("Choose New Initial State: ");
        initialStateComboBox = new JComboBox<String>(States);

        panel.add(originLabel);
        panel.add(initialStateComboBox);

        return panel;
    }

    public static void main(String[] args){
        SwingUtilities.invokeLater(new Runnable() {
            @Override
            public void run() {
                new SetInitialStateDialog(new String[]{"q0","q1"}).displayGUI();
            }
        });
    }
}

File: src/AutomatonTheory/Kotlin/Swing/DialogBoxes/SetInitialStateDialog.kt
package AutomatonTheory.Kotlin.Swing.DialogBoxes

import javax.swing.*

class SetInitialStateDialog(var States: Array<String>) {

    var initialStateComboBox = JComboBox<String>()
    var newInitialState = ""
    var valor: Int = 0
    fun displayGUI() {
        valor = JOptionPane.showConfirmDialog(null,
                panel,
                "Set Initial State",
                JOptionPane.OK_CANCEL_OPTION,
                JOptionPane.PLAIN_MESSAGE)
        newInitialState = initialStateComboBox.selectedItem.toString()
    }

    private val panel: JPanel
        get() {

            val panel = JPanel()
            panel.layout = BoxLayout(panel, BoxLayout.Y_AXIS)
            val originLabel = JLabel("Choose New Initial State: ")
            initialStateComboBox = JComboBox(States)

            panel.add(originLabel)
            panel.add(initialStateComboBox)

            return panel
        }

    companion object {

        @JvmStatic fun main(args: Array<String>) {
            SwingUtilities.invokeLater { SetInitialStateDialog(arrayOf("q0", "q1")).displayGUI() }
        }
    }
}
